test/functional/normalization/json_parsing_test.go

```diff
@@ -0,0 +1,177 @@
+		expectedMessage := jsonMsg
+		jsonMsg = strings.ReplaceAll(jsonMsg, "\"", "\\\"")
+		timestamp := "2020-11-04T18:13:59.061892+00:00"
+
+		// Write log line as input to fluentd
+		applicationLogLine := fmt.Sprintf("%s stdout F %s", timestamp, jsonMsg)
+		Expect(framework.WriteMessagesToApplicationLog(applicationLogLine, 10)).To(BeNil())
+
+		// Read line from Log Forward output
+		raw, err := framework.ReadApplicationLogsFrom(logging.OutputTypeFluentdForward)
+		Expect(err).To(BeNil(), "Expected no errors reading the logs")
+
+		// Parse log line
+		var logs []types.ApplicationLog
+		err = types.StrictlyParseLogs(raw, &logs)
+		Expect(err).To(BeNil(), "Expected no errors parsing the logs")
+		Expect(logs[0].Structured).To(Equal(expected), "parsed json message not matching")
+		Expect(logs[0].Message).To(Equal(expectedMessage), "received message not matching")
+	})
+	It("should not parse non json message into structured", func() {
+		clfb.Forwarder.Spec.Pipelines[0].Parse = "json"
+		Expect(framework.Deploy()).To(BeNil())
+
+		// Log message data
+		message := "Functional test message"
+		timestamp := "2020-11-04T18:13:59.061892+00:00"
+
+		// Write log line as input to fluentd
+		applicationLogLine := fmt.Sprintf("%s stdout F %s", timestamp, message)
+		Expect(framework.WriteMessagesToApplicationLog(applicationLogLine, 10)).To(BeNil())
+
+		// Read line from Log Forward output
+		raw, err := framework.ReadApplicationLogsFrom(logging.OutputTypeFluentdForward)
+		Expect(err).To(BeNil(), "Expected no errors reading the logs")
+
+		// Parse log line
+		var logs []types.ApplicationLog
+		err = types.StrictlyParseLogs(raw, &logs)
+		Expect(err).To(BeNil(), "Expected no errors parsing the logs")
+		Expect(logs[0].Structured).To(Equal(map[string]interface{}{}), "received non-empty json parsed structure")
+		Expect(logs[0].Message).To(Equal(message), "received message not matching")
+	})
+	XIt("should not parse invalid json message into structured", func() {
+		clfb.Forwarder.Spec.Pipelines[0].Parse = "json"
+		Expect(framework.Deploy()).To(BeNil())
+
+		// Log message data
+		message := `{"key":"v}`
+		timestamp := "2020-11-04T18:13:59.061892+00:00"
+
+		// Write log line as input to fluentd
+		message = strings.ReplaceAll(message, "\"", "\\\"")
+		applicationLogLine := fmt.Sprintf("%s stdout F %s", timestamp, message)
+		Expect(framework.WriteMessagesToApplicationLog(applicationLogLine, 10)).To(BeNil())
+
+		// Read line from Log Forward output
+		raw, err := framework.ReadApplicationLogsFrom(logging.OutputTypeFluentdForward)
+		Expect(err).To(BeNil(), "Expected no errors reading the logs")
+
+		// Parse log line
+		var logs []types.ApplicationLog
+		err = types.StrictlyParseLogs(raw, &logs)
+		Expect(err).To(BeNil(), "Expected no errors parsing the logs")
+		Expect(logs[0].Structured).To(Equal(map[string]interface{}{}), "received structured not matching")
+		Expect(logs[0].Message).To(Equal(message), "received message not matching")
+	})
+
```

Comment:
Maybe add one case to make sure that we don't parse valid json if Parse is not set, i.e. we default to original behavior.

Reply:
Added